### PR TITLE
Add types for Suspense/lazy

### PR DESCRIPTION
**Pull request by pmkroeker**
This adds some types to the newly merged Suspense/lazy!

Currently Suspense is an extension of Component, but I think it should likely be a bit more rigid than that.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -230,5 +230,17 @@ declare namespace preact {
 	}
 	interface PreactContext<T> extends Context<T> {}
 
-	function createContext<T>(defaultValue: T): Context<T>;
+  function createContext<T>(defaultValue: T): Context<T>;
```
Comment (andrewiggins) [inline src/index.d.ts]: The spacing of these lines seems off from the rest of the file (note, I think we use tabs in our src). If there is an editorconfig extension for your editor, using that will just do the right thing for you

Comment (pmkroeker) [reply src/index.d.ts]: Ya I think I have mine configured to 2 spaces. I'll fix this.

Comment (pmkroeker) [reply src/index.d.ts]: Ok, took a bit to figure out the correct config, but I think I got it setup properly now.

**Comment by pmkroeker** [comment]
Yes I was just thinking that I should probably add some tests. Was looking through the TS tests section and it looks like some of the tests are from a while ago (poss pre-X). I'll add in some now!

**Comment by pmkroeker** [comment]
Added some basic tests. Had to mimic `import('./module')` as it threw an error on `import`.

File: test/ts/suspense-test.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import "mocha";
+import { expect } from "chai";
+import {
+	createElement,
+	Component,
+        Suspense,
+        lazy,
+        AnyComponent
+} from "../../src/";
```
Comment (ForsakenHarmony) [inline test/ts/suspense-test.tsx]: Same here

File: test/ts/lazy.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import {
+	createElement, Component,
+} from "../../src/";
```
Comment (ForsakenHarmony) [inline test/ts/lazy.tsx]: This could probably be a `".. /.." `?

File: test/ts/lazy.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+//                         <div>{
+//                                 isProp ?
+//                                 'Super Lazy TRUE' :
+//                                 'Super Lazy FALSE'
+//                         }</div>
+//                 )
+//         }
+// }
+
+export const IsLazyComponent = (props: LazyProps) =>
```
Comment (ForsakenHarmony) [inline test/ts/lazy.tsx]: Could probably remove the commented out code? And potentially add a type annotation here, but that's not important

File: test/ts/suspense-test.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import { IsLazyComponent } from './lazy';
+
+const FallBack = () => <div>Still working...</div>;
+/**
+ * Have to moch dynamic import as import() throws a syntax error in the test runner
```
Comment (ForsakenHarmony) [inline test/ts/suspense-test.tsx]: "mock" * 🙃

Comment (pmkroeker) [reply test/ts/suspense-test.tsx]: Haha whoops, going to go through things for a cleanup tomorrow.

File: test/ts/lazy.tsx (hunk 3 above)
Comment (pmkroeker) [reply test/ts/lazy.tsx]: Done!

File: test/ts/lazy.tsx (hunk 4 above)
Comment (pmkroeker) [reply test/ts/lazy.tsx]: I'm actually going to un-comment this section and have examples of both.

File: test/ts/suspense-test.tsx (hunk 2 above)
Comment (pmkroeker) [reply test/ts/suspense-test.tsx]: Done!

File: test/ts/suspense-test.tsx (hunk 5 above)
Comment (pmkroeker) [reply test/ts/suspense-test.tsx]: Fixed!

File: src/index.d.ts [hunk 6]
```diff
@@ -230,5 +230,17 @@ declare namespace preact {
 	}
 	interface PreactContext<T> extends Context<T> {}
 
-	function createContext<T>(defaultValue: T): Context<T>;
+        function createContext<T>(defaultValue: T): Context<T>;
```
[the remark this answers is not in the document]
Comment (pmkroeker) [reply src/index.d.ts]: Fixed, took a bit to sort out my editor config. Fixed all the white spacing in the other files as well.

**Comment by pmkroeker** [comment]
I realized that I typed lazy incorrectly arg should be `() => Promise<...>`. This has unfortunately led to some previous typings not quite working as expected. Working on a fix now.

**Comment by pmkroeker** [comment]
With the current type of `lazy` its usage is:
```typescript
const IsLazy = lazy<typeof import('./lazy'), LazyProps>(() => import('./lazy'));
```
Which seems pretty verbose. Was hoping to get a more streamlined approach.

**Comment by pmkroeker** [comment]
Alright I was over complicating the typings a bit, simplifying lazy to 
```typescript
function lazy<T>(loader: () => Promise<{default: T}>): T;
```
Solved the issue with IMO a verbose usage to the preferred:
```typescript
const IsLazy = lazy(() => import('./lazy'));
```

And will still typecheck the components props properly.

**Comment by pmkroeker** [comment]
My pleasure!

**Pull request merged**